#[facet(default)] with lifetime can not be compiled

```rust
// this can not be compiled
// report: error[E0401]: can't use generic parameters from outer item
#[derive(Facet)]
struct Foo<'a> {
    #[facet(default)]
    name: Option<Cow<‘a, str>>,
}
```

```rust
// but this ok
#[derive(Facet)]
struct Foo<'a> {
    #[facet(default)]
    name: Option<String>,
    name2: Cow<'a, str>,
}
```

--- a/facet-derive-emit/src/process_struct.rs
+++ b/facet-derive-emit/src/process_struct.rs
@@ -12,6 +12,18 @@ pub(crate) fn gen_field_from_pfield(
     let field_name_raw = &field.name.raw;
     let field_type = &field.ty; // TokenStream of the type
 
+    let tts: facet_derive_parse::TokenStream = field_type.clone();
+    let field_type_static = tts
+        .to_token_iter()
+        .parse::<Vec<LifetimeOrTt>>()
+        .unwrap()
+        .into_iter()
+        .map(|lott| match lott {
+            LifetimeOrTt::TokenTree(tt) => quote! { #tt },
+            LifetimeOrTt::Lifetime(_) => quote! { 'static },
+        })
+        .collect::<TokenStream>();
+
     let bgp_without_bounds = bgp.display_without_bounds();
 
     // Determine field flags and other attributes from field.attrs
@@ -43,7 +55,7 @@ pub(crate) fn gen_field_from_pfield(
                     flags = quote! { #flags.union(::facet::FieldFlags::DEFAULT) };
                 }
                 asserts.push(quote! {
-                    ::facet::static_assertions::assert_impl_all!(#field_type: ::core::default::Default);
+                    ::facet::static_assertions::assert_impl_all!(#field_type_static: ::core::default::Default);
                 })
             }
             PFacetAttr::DefaultEquals { expr } => {

--- a/facet-derive-emit/tests/codegen/mod.rs
+++ b/facet-derive-emit/tests/codegen/mod.rs
@@ -1091,3 +1091,16 @@ fn enum_with_macro_discriminants() {
         "#
     ));
 }
+
+#[test]
+fn struct_with_option_cow_str() {
+    insta::assert_snapshot!(expand(
+        r#"
+        #[derive(Facet)]
+        struct Foo<'a> {
+            #[facet(default)]
+            name: Option<Cow<'a, str>>,
+        }
+        "#
+    ));
+}

--- a/facet-derive-parse/src/lib.rs
+++ b/facet-derive-parse/src/lib.rs
@@ -472,6 +472,14 @@ unsynn! {
         /// The fields enclosed in braces `{}`.
         pub fields: BraceGroupContaining<CommaDelimitedVec<StructField>>,
     }
+
+    /// A lifetime or a tokentree, used to gather lifetimes in type definitions
+    pub enum LifetimeOrTt {
+        /// A lifetime annotation.
+        Lifetime(Lifetime),
+        /// A single token tree.
+        TokenTree(TokenTree),
+    }
 }
 
 impl core::fmt::Display for AngleTokenTree {

--- a/facet/tests/derive.rs
+++ b/facet/tests/derive.rs
@@ -537,3 +537,12 @@ fn core_ops_range() {
         offset_of!(core::ops::Range::<usize>, end)
     );
 }
+
+#[test]
+fn struct_with_default_field_that_has_lifetime() {
+    #[derive(Facet)]
+    struct Foo<'a> {
+        #[facet(default)]
+        name: Option<std::borrow::Cow<'a, str>>,
+    }
+}

